script.js

```diff
@@ -55,18 +55,38 @@ const form = document.querySelector('.form');
 const email = document.querySelector('#email');
 const error = document.querySelector('.error');
 
-function checkEmail(Diana) {
-  if (Diana.match(/^[a-z@.0-9-_]*$/)) {
+function checkEmail(emailAddress) {
+  if (emailAddress.match(/^[a-z@.0-9-_]*$/)) {
     return true;
   }
   return false;
 }
 
+const userid = document.getElementById('name');
+const em = document.getElementById('email');
+const txt = document.getElementById('message');
+
+function storeData() {
+  const testObject = {
+    name: userid.value,
+    email: email.value,
+    message: txt.value,
+  };
+  localStorage.setItem('store', JSON.stringify(testObject));
+}
+
+const emptyObject = JSON.parse(localStorage.getItem('store'));
+
+userid.value = emptyObject.name;
+em.value = emptyObject.email;
+txt.value = emptyObject.message;
+
 form.addEventListener('submit', (e) => {
```

Comment:
- [ ] You are yet to implement the storage as explained by the previous reviewer. You can see that your data only saves after form submission. You still need to handle the scenario when the user does not submit it.

- [ ] Your data does not seem to be loaded and presented when the page loads.

  For both cases above I would propose this code as it seems to resolve both issues. This loads the data from the local storage only after the DOM has loaded then changing `submit` to `change` ensures that the local storage is updated once you change focus on the form inputs

  ```
  document.addEventListener('DOMContentLoaded', () => {
    const emptyObject = JSON.parse(localStorage.getItem('store'));
    
    userid.value = emptyObject.name;
    em.value = emptyObject.email;
    txt.value = emptyObject.message;
  })
  
  form.addEventListener('change', (e) => {
    ...
  ```